climpyrical/cmd/preprocess_model.py

```diff
@@ -0,0 +1,169 @@
+
+import warnings
+
+import numpy as np
+
+warnings.filterwarnings("ignore")
+
+
+@click.command()
+@click.option("-i", "--in-path", help="Input CanRCM4 file", required=True)
+@click.option("-o", "--out-path", help="Output file", required=True)
+@click.option(
+    "-m", "--fill-glaciers", help="Refill glacier points", default=True
+)
+@click.option(
+    "-l",
+    "--log-level",
+    help="Logging level",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+    default="INFO",
+)
+def run_processing(in_path, out_path, fill_glaciers, log_level):
+    """Completes the preprocessing of the model required
+    for the NRC project.
+    Args:
+        in_path, out_path (strings): directories of NetCDF4 file
+            input and output. Must give filename, too with extension
+            .nc. Overwites files with same name in same directory.
+        fill_glaciers (bool): whether to fill spurious glacier
+            points with preprocessed mask. Default is True.
+        quiet (bool): whether to log quietly or not
+    Returns:
+        Creates a NetCDF4 file at out_path at target resolution
+    """
+    logging.basicConfig(level=log_level)
+
+    ds = read_data(in_path)
+    dv = list(ds.data_vars)[0]
+
+    rlon, rlat = np.meshgrid(ds.rlon, ds.rlat)
+    mean = ds[dv].values
+    kelvin = 273.15  # K
+
+    logging.info("Detect unit conversions")
+    if dv in ["twb", "tas"]:
+        print("Temperature field detected. Converting to Kelvin.")
+        mean += kelvin
+
+    path_mask = resource_filename(
+        "climpyrical", "nrc_data/land_mask_CanRCM4_sftlf.nc"
+    )
+
+    path_glacier_mask = resource_filename(
+        "climpyrical", "nrc_data/glacier_mask.nc"
+    )
+
+    logging.info("Load and regrid file to target resolution")
+    mask = read_data(path_mask)
+    mask = regrid_ensemble(mask, "sftlf", 10, copy=True)
+    mask = mask["sftlf"] >= 1.0
+
+    logging.info("Load original reoslution mask for reference")
+    mask_og = read_data(path_mask)["sftlf"].values != 0.0
+
+    glaciermask = read_data(path_glacier_mask)["mask"].values != 0.0
+
+    logging.info(
+        "Insert NaN values into glacier points to fill"
+        "and interpolate if fill_galciers is set"
+    )
+    if fill_glaciers:
+        logging.info("Filling spurious glacier points.")
+        mean[glaciermask] = np.nan
+
+    nanmask = ~np.isnan(mean)
+    points = np.stack([rlon[nanmask], rlat[nanmask]]).T
+    target_values = mean[nanmask]
+    target_points = np.stack([rlon[glaciermask], rlat[glaciermask]]).T
+
+    mean[glaciermask] = interpolate_dataset(
+        points, target_values, target_points, "linear"
+    )
+
+    ds = gen_dataset(dv, mean, ds.rlat, ds.rlon, ds.lat, ds.lon)
+    ds = ds.assign({dv: (["rlat", "rlon"], mean)})
+
+    logging.info("Remove water cells at original resolution")
+    ds[dv].values[~mask_og] = np.nan
+    nanmask = ~np.isnan(ds[dv].values)
+
+    ds10 = regrid_ensemble(ds, dv, 10, copy=True)
```

Comment:
`ds10`? The intent here isn't obvious to me.

Reply:
`ds10` is  `ds` but copied and downscaled 10x in each coordinate axis.

Reply:
Don't tell me, tell future readers of the code ;)